<launch>
    <!-- Define arguments with default values -->
    <arg name="namespace" default="1"/>   
    <arg name="camera_name" default="1"/>
    <arg name="camera_ID" default="1"/>
    <arg name="subscriber_name" default="1"/>
    <arg name="image_topic" default="1"/>
    <arg name="detection_topic" default="1"/>

    <group ns="$(arg namespace)">

        <node pkg="libuvc_camera" type="camera_node" name="cam1">
            <param name="vendor" value="0x32e4"/>
            <param name="product" value="0x9422"/>
            <param name="serial" value=""/>
            <param name="index" value="$(arg camera_ID)"/>
            <param name="width" value="1280"/>
            <param name="height" value="720"/>
            <param name="video_mode" value="mjpeg"/>
            <param name="frame_rate" value="30"/>
        </node>

        <!-- <node name="image_flipper" pkg="testpackage" type="image_flipper.py" args="image_raw image_flipped" output="screen"/> -->

        <node name="yolov5_node" pkg="testpackage" type="yolov5_node.py" output="screen" args="image_raw $(arg detection_topic)"/>

        <node name="image_subscriber" pkg="testpackage" type="image_subscriber.py" args="$(arg detection_topic)" output="screen">
            <param name="topic_name" type="string" value="$(arg namespace)/$(arg detection_topic)" />
        </node>


    </group>
</launch>